examples/llm_plus_gnn/glem.py

```diff
@@ -0,0 +1,372 @@
+                else:
+                    best_val_acc = val_acc
+                    final_test_acc = test_acc
+                    preds = cur_preds
+
+        if em_phase == 'gnn':
+            gnn_test_acc = max(gnn_test_acc, final_test_acc)
+            # torch.save(preds, osp.join(preds_dir, f'lm.pt'))
+            # del model.gnn
+            model.gnn.cpu()
+            em_phase = 'lm'
+        else:
+            lm_test_acc = max(lm_test_acc, final_test_acc)
+            # torch.save(preds, osp.join(preds_dir, f'gnn.pt'))
+            # del model.lm
+            model.lm.cpu()
+            em_phase = 'gnn'
+        torch.cuda.empty_cache()
+    print(f'Best GNN acc: {gnn_test_acc}, LM acc: {lm_test_acc}')
+    print('============================')
+    end_time = time.time()
+    running_time = (end_time - start_time) / 3600
+    print(f'Total running time: {running_time:.2f} hours')
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='GLEM Example:')
+    parser.add_argument('--gpu', type=int, default=0)
+    parser.add_argument('--num_runs', type=int, default=10,
+                        help='number of runs')
+    parser.add_argument('--num_em_iters', type=int, default=1,
+                        help='number of iterations')
+    parser.add_argument("--dataset", type=str, default='arxiv',
+                        help='arxiv or products')
+    parser.add_argument("--pl_ratio", type=float, default=0.5,
+                        help="pseudo labels ratio")
+    parser.add_argument('--hf_model', type=str, default='prajjwal1/bert-tiny',
+                        help='huggingface model repo id')
+    parser.add_argument(
+        '--gnn_model', type=str, default='SAGE',
+        help='gnn model for node classification,'
+        'options: SAGE, GAT, GCN')
+    parser.add_argument('--lm_batch_size', type=int, default=256)
+    parser.add_argument('--gnn_batch_size', type=int, default=1024)
+    parser.add_argument('--external_pred_path', type=str, default=None)
+    parser.add_argument('--alpha', type=float, default=0.5,
+                        help='pseudo label weight in E-step')
+    parser.add_argument('--beta', type=float, default=0.5,
+                        help='pseudo label weight in M-step')
+    parser.add_argument('--lm_epochs', type=int, default=10)
+    parser.add_argument('--gnn_epochs', type=int, default=50)
+    parser.add_argument('--gnn_lr', type=float, default=0.003)
+    parser.add_argument('--lm_lr', type=float, default=0.001)
+    parser.add_argument('--patience', type=int, default=5)
+    parser.add_argument('--verbose', action='store_true',
+                        help='show progress bar during training or not')
+    parser.add_argument('--em_order', type=str, default='lm')
+    parser.add_argument('--lm_use_lora', action='store_true',
+                        help='use Lora to fine-tune model or not')
+    parser.add_argument('--token_on_disk', action='store_true')
```

Comment:
add help to explain that this is used to save the tokens on disk in case RAM is insufficient